pkg/resourcemanager/config/operatormode.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// OperatorMode determines whether we'll run watchers and/or webhooks.
+type OperatorMode int
+
+const (
+	OperatorModeWatchers = OperatorMode(1 << iota)
+	OperatorModeWebhooks
+
+	OperatorModeBoth = OperatorModeWatchers | OperatorModeWebhooks
+)
+
+// IncludesWebhooks returns whether an operator running in this mode
+// should register webhooks.
+func (m OperatorMode) IncludesWebhooks() bool {
+	return m&OperatorModeWebhooks > 0
+}
+
+// IncludesWatchers returns whether an operator running in this mode
+// should register reconcilers.
+func (m OperatorMode) IncludesWatchers() bool {
+	return m&OperatorModeWatchers > 0
+}
+
+// String converts the mode into a readable value.
+func (m OperatorMode) String() string {
+	switch m {
+	case OperatorModeWatchers:
+		return "watchers"
+	case OperatorModeWebhooks:
+		return "webhooks"
+	case OperatorModeBoth:
```

Comment:
Read in context, `operatorMode=both` is maybe a bit weird in that it begs the question "both what"?

This also assumes there shall be no further modes, as if there are `both` becomes confusing. What does it mean when there are 3 modes?

Wondering if we want `both` to instead be like `webhooksAndWatchers` or `webhooks|watchers` or something?

Reply:
Yeah, fair point - how about `webhooks-and-watchers`? I'd rather it was all lowercase, and using bitwise-or makes it read kind of the opposite from the actual meaning.

Reply:
And it's ok if it's verbose since it's the default.